src/celeritas/em/distribution/UrbanMscScatter.hh

```diff
@@ -201,8 +201,10 @@ CELER_FUNCTION auto UrbanMscScatter::operator()(Engine& rng) -> MscInteraction
     if (skip_sampling_)
     {
         // Do not sample scattering at the last or at a small step
-        return {
-            true_path_, {0, 0, 0}, {0, 0, 0}, MscInteraction::Action::unchanged};
+        return {true_path_,
+                inc_direction_,
```

Comment:
What failed when the exiting direction was set to zero? Since the action is unchanged, nothing should have touched it.

Reply:
If my memory serves me right, the momentum x-direction was 0 (should be 1) for 100 MeV when {0,0,0} is returns.  I do not find where the side effect happens as the logic should be fine as long as the result is not updated for Action::unchanged.  Nevertheless, the original direction should be used as it is conceptually more correct.

Reply:
Agreed, it's conceptually more correct, and the return signature here was laziness on my part. However, if the return direction is actually used elsewhere, it indicates an error in the code... and if it's in error but a test isn't currently failing, then that indicates a hole in the testing.